addons/repair/models/repair.py

```diff
@@ -497,6 +495,7 @@ def action_repair_end(self):
         if self.filtered(lambda repair: repair.state != 'under_repair'):
             raise UserError(_("Repair must be under repair in order to end reparation."))
         for repair in self:
+            repair._check_product_tracking()
```

Comment:
would be nice to have this check before the loop `for repair in self` to not make less process before having a potential error. you can even call it with the whole recordset